### PR TITLE
[WIP] Add option to clean scheduled messages on startup

### PR DESCRIPTION
**Pull request by kenliao94**
Please check if JMS scheduler is not used for other things than sending messages. I couldn't find any other usecases.

I also don't really like how i need to expose the internal name of the job scheduler of the broker. But then removing all scheduled messages is in the persistent layer. The current approach seems pretty neat and lightweight. Nevertheless, I am not aware of all the edge cases yet.

I am going to write tests if the approach looks good.

### CODE REVIEW

File: activemq-broker/src/main/java/org/apache/activemq/broker/BrokerService.java [hunk 1]
```diff
@@ -699,6 +700,10 @@ private void doStartPersistenceAdapter() throws Exception {
         if (jobSchedulerStore != null) {
             try {
                 jobSchedulerStore.start();
+                if (deleteAllScheduledMessagesOnStartup) {
+                    LOG.info("Deleting all scheduled messages on startup because deleteAllScheduledMessagesOnStartup configuration has been provided");
+                    deleteAllScheduledMessages();
```
Comment (Nikita-Shupletsov) [inline activemq-broker/src/main/java/org/apache/activemq/broker/BrokerService.java]: probably, it makes sense to do something similar to deleteAllMessages: https://github.com/apache/activemq/blob/main/activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java#L495
when the flag is true, we just delete everything(the journal and the index) in the load method

File: activemq-broker/src/main/java/org/apache/activemq/broker/BrokerService.java [hunk 2]
```diff
@@ -1630,6 +1635,24 @@ public void setDeleteAllMessagesOnStartup(boolean deletePersistentMessagesOnStar
...
+    }
+
+    /**
+     * Sets whether or not all scheduled messages are deleted on startup
+     * @org.apache.xbean.Property propertyEditor="org.apache.activemq.util.BooleanEditor"
+     */
+    public void setDeleteAllScheduledMessagesOnStartup(boolean deleteAllScheduledMessagesOnStartup) {
+        this.deleteAllScheduledMessagesOnStartup = deleteAllScheduledMessagesOnStartup;
+    }
+
```
Comment (pa-deasy) [inline activemq-broker/src/main/java/org/apache/activemq/broker/BrokerService.java:1641]: The `addInterceptors` method calls `sb.setMaxRepeatAllowed(maxSchedulerRepeatAllowed);`.
Would it be possible to also expose a `setDeleteAllScheduledMessagesOnStartup` method on `SchedulerBroker`? Then move the delete functionality into  `SchedulerBroker` and invoke it on `start()` if `deleteAllScheduledMessagesOnStartup` is true.

Comment (kenliao94) [reply activemq-broker/src/main/java/org/apache/activemq/broker/BrokerService.java:1641]: That's a good idea actually. I took a brief look before and my original concern is should the scheduler be aware of messages or it should be generic to job. But taking a second look it seems like it has message related operation there, so I will use this approach. And then I will raise a draft PR in upstream asking for maintainer opinion.

File: activemq-broker/src/main/java/org/apache/activemq/broker/BrokerService.java (hunk 1 above)
Comment (kenliao94) [reply activemq-broker/src/main/java/org/apache/activemq/broker/BrokerService.java]: I think the removeAllJobs does the similar thing?
